src/webdav4/client.py

```diff
@@ -502,7 +507,11 @@ def ls(  # pylint: disable=invalid-name
                 (non-collection), ls will return the file entry/details.
                 Otherwise, it will raise an error.
         """
-        result = self.propfind(path, headers={"Depth": "1"})
+        result = self.propfind(
```

Comment:
`ls` supports listing the resource as well, which AFAIK should be sent without a trailing slash. The only solution
that I can think of is either:
1. support `follow_redirects` in `propfind` or,
2. do an `ls` call on the parent and take an appropriate decision on adding a trailing slash based on that. I have tried to avoid unnecessary calls as much as possible in `webdav4` so I don't like this approach. :)

Reply:
I'll revise the PR to follow_redirects instead.

Reply:
Done in https://github.com/skshetry/webdav4/pull/103/commits/0e6613cf4274930a71ec26f0ec123af87467c741